File: inst/debug/simple-addition.py
from inspect_ai import Task, task
from inspect_ai.dataset import Sample
from inspect_ai.scorer import model_graded_qa
from inspect_ai.solver import generate


@task
def simple_addition():
    dataset = [
        Sample(input="What's 2+2?", target="4"),
        Sample(input="What's 2+3?", target="5"),
        Sample(input="What's 2+4?", target="6"),
    ]

    return Task(
        dataset=dataset,
        solver=generate(),
        scorer=model_graded_qa(),
        model="anthropic/claude-sonnet-4-20250514",
    )

File: inst/debug/simple-addition.R
library(vitals)

vitals_log_dir_set("inst/debug/vitals/")

simple_addition <- tibble(
  input = c("What's 2+2?", "What's 2+3?", "What's 2+4?"),
  target = c("4", "5", "6")
)

tsk <- Task$new(
  dataset = simple_addition,
  solver = generate(chat_claude(model = "claude-sonnet-4-5")),
  scorer = model_graded_qa()
)

tsk$eval()
